### ISSUE TITLE
Decimal.cmp/2 is deprecated. Use compare/2 instead

### ISSUE THREAD

**Issue by inoas** [elixir-ecto/ecto#3470]
### Environment

* Elixir version (elixir -v): `Erlang/OTP 23 [erts-11.1.1] [source] [64-bit] [smp:16:16] [ds:16:16:10] [async-threads:1] [hipe] [dtrace]`

```
warning: Decimal.cmp/2 is deprecated. Use compare/2 instead
  lib/ecto/changeset.ex:2155: Ecto.Changeset.validate_number/6
```

**Comment by josevalim** [comment]
Correct. We are keeping it for backwards compatibility but we will drop it soon. Thanks!

**Comment by v0idpwn** [comment]
This should be closed due to #3504

**Comment by wojtekmach** [comment]
Thanks @v0idpwn!
